public/apps/login/login-page.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { Component } from 'react';
+import { EuiText, EuiFieldText, EuiIcon, EuiSpacer, EuiButton, EuiImage } from '@elastic/eui';
+  
+class LoginPageProps {
+    appBasePath: string;
+}
+
+class LoginPageStates {
+    username: string;
+    password: string;
+    loginFailed: boolean;
+}
+
+export class LoginPage extends Component<LoginPageProps, LoginPageStates> {
+    constructor(props: LoginPageProps) {
+        super(props);
+        this.state = {username: "", password: "", loginFailed: false}
+    }
+
+    onUsernameChange = (e) => {
```

Comment:
I prefer this naming convention :

Passing as props name `onEvent` prefix while actual implementation has `handleEvent`

```
class ComponentA {
    handleEvent() {}
    render() {
        return(<ComponentB onclick={handleEvent}/>);
}
```